### PR TITLE
feat: API endpoint to see how many alerts a provider has sent

### LINKED ISSUE #1249: [➕ Feature] API endpoint to see how many alerts a provider has sent
This endpoint could be then used to monitor the "health" of some provider or just expose some BI of trends.

**Describe your use case**
An API endpoint that exposes the number of alerts a specific provider has received (in a specific time time period or ever)

**Are you already using Keep?**
**Yes**/No

**Additional context**
-

### PR DESCRIPTION
**Pull request by 35C4n0r**
- add service function in api/core/db.py
- add endpoint in api/routes/providers.py

Closes #1249 
/claim #1249

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: keep/api/routes/providers.py [hunk 1]
```diff
@@ -229,6 +229,35 @@ def get_alerts_schema(
...
+    ever: bool,
+    start_time: Optional[datetime.datetime] = None,
+    end_time: Optional[datetime.datetime] = None,
+    authenticated_entity: AuthenticatedEntity = Depends(AuthVerifier(["read:alert"])),
+):
+    tenant_id = authenticated_entity.tenant_id
+    if ever is False and (start_time is None or end_time is None):
+        return JSONResponse(
+            status_code=400, content={"message": "Missing start_time or end_time"}
+        )
```
Comment (talboren) [inline keep/api/routes/providers.py:245]: raise HTTPException instead

File: keep/api/routes/providers.py [hunk 2]
```diff
@@ -229,6 +229,35 @@ def get_alerts_schema(
...
+        status_code=200,
+        content={
+            "alert_count": count_alerts(
+                provider_type=provider_type,
+                provider_id=provider_id,
+                start_time=start_time,
+                end_time=end_time,
+                ever=ever,
+                tenant_id=tenant_id,
+            )
```
Comment (talboren) [inline keep/api/routes/providers.py:256]: lets define a model for it

File: keep/api/core/db.py [hunk 3]
```diff
@@ -733,6 +733,39 @@ def enrich_alert(
...
 
 
+def count_alerts(
+    ever: bool,
+    start_time: Optional[datetime],
+    end_time: Optional[datetime],
+    provider_id: str,
+    provider_type: str,
+    tenant_id: str,
+):
```
Comment (talboren) [inline keep/api/core/db.py:743]: add documentation please

File: keep/api/routes/providers.py (hunk 2 above)
Comment (talboren) [reply keep/api/routes/providers.py:256]: (e.g., ProviderAlertsCountDto)

**Review by talboren — CHANGES_REQUESTED** [review]
some small changes

**Comment by talboren** [comment]
@35C4n0r thank you for the contribution! left some small comments, can you please take a look at them?

**Comment by 35C4n0r** [comment]
Hey @talboren, thanks for the review. I've made the needed changes.

File: keep/api/routes/providers.py [hunk 4]
```diff
@@ -229,6 +230,40 @@ def get_alerts_schema(
...
+                    provider_type=provider_type,
+                    provider_id=provider_id,
+                    ever=ever,
+                    start_time=start_time,
+                    end_time=end_time,
+                    tenant_id=tenant_id,
+                ),
+            )
+        },
+    )
```
Comment (talboren) [inline keep/api/routes/providers.py:264]: JSONResponse is redundant

File: keep/api/models/provider.py [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
 from keep.providers.models.provider_method import ProviderMethod
 
 
+class ProviderAlertsCountDTO(BaseModel):
+    provider_type: str
+    provider_id: str
+    ever: bool
+    start_time: Optional[datetime]
+    end_time: Optional[datetime]
+    tenant_id: str
```
Comment (talboren) [inline keep/api/models/provider.py:16]: I meant to use this for the response model not for the request model 😅

File: keep/api/core/db.py [hunk 6]
```diff
@@ -38,6 +38,7 @@
 from keep.api.models.db.rule import *  # pylint: disable=unused-wildcard-import
 from keep.api.models.db.tenant import *  # pylint: disable=unused-wildcard-import
 from keep.api.models.db.workflow import *  # pylint: disable=unused-wildcard-import
+from keep.api.models.provider import ProviderAlertsCountDTO
```
Comment (talboren) [inline keep/api/core/db.py:41]: better not import dtos in here, return the .count() number and construct the response in the route itself

**Review by talboren — CHANGES_REQUESTED** [review]
A few more changes 🙏🏼

**Review by talboren — APPROVED** [review]
LGTM

**Pull request merged**